### DISCUSSION TITLE
[solution] System.DllNotFoundException: Unable to load shared library 'libSkiaSharp' or one of its dependencies.

### DISCUSSION THREAD

**Discussion by lukiies** [AvaloniaUI/Avalonia#5525] [Q&A]
Hi everyone,
what i'm trying to do is preparing new development env from scratch on linux:
- install ubuntu server + ubuntu desktop + essentials..
- install .net 5.0 sdk + runtime 50 via snap
- add some needed env : export DOTNET_ROOT=/snap/dotnet-sdk/current && export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1 to /etc/profile
- rider preview version
- install dotnet new -i Avalonia.Templates
- create mvvm project from templates (in/out rider)
- debug run

catch error like this:

System.DllNotFoundException: Unable to load shared library 'libSkiaSharp' or one of its dependencies.

I just started consider avalonia, but think probably everyone can do this the same way and stuck at the begining.

very simple SOLUTION is here:

- avalonia.desktop 0.10 dependency need SkiaSharp.NativeAssets.Linux, which need fontconfig native lib - but i red isnt important for avalonia  &  can be changed easly into SkiaSharp.NativeAssets.Linux.NoDependentions
- during build libSkiaSharp.so is copied from ~/.nuget/packages/skiasharp.nativeassets.linux/2.80.2/runtimes/linux-x64/native to <project-dir>/bin/Debug/net5.0/runtimes/linux-x64/native

project doesnt run this native library isnt working in such as prepared environment - BUT you can do:

`cp ~/.nuget/packages/skiasharp.nativeassets.linux.nodependencies/2.80.3-preview.40/runtimes/linux-x64/native/libSkiaSharp.so <project-dir>/bin/Debug/net5.0`

and this projekt will run without problems.

BUT if you create next new project mvvm - you will have the same bug of course, because during building this native lib is copied automatically into <project-dir>/bin/Debug/net5.0/runtimes/linux-x64/native
temporary but working solution is to:

`cp ~/.nuget/packages/skiasharp.nativeassets.linux.nodependencies/2.80.3-preview.40/runtimes/linux-x64/native/libSkiaSharp.so ~/.nuget/packages/skiasharp.nativeassets.linux/2.80.2/runtimes/linux-x64/native`

every new created project mvvm will then run without those problems..

of course very ugly solution, i doesnt want to have something like that in my environment
does everybody know how to repair build system proper way ?

thx

**Comment by kekekeks** [comment]
You can exclude binaries supplied by SkiaSharp.NativeAssets.Linux by adding 

```xml
<ItemGroup>
  <PackageReference Include="SkiaSharp.NativeAssets.Linux" Version="2.80.2" ExcludeAssets="all" />
</ItemGroup>
```

**Reply by kekekeks** [reply]
Also note that libfontconfig dependency *is important* unless you are bundling all of your fonts as resources.

**Reply by lukiies** [reply]
thx a lot - this is what im looking for

how can i use this library or check why it cant run ?
(ldd shows everything is fine with libSkiaSharp.so)

**Reply by kekekeks** [reply]
1) try setting LD_LIBRARY_PATH
2) try without that snap abomination

**Reply by lukiies** [reply]
LD_LIBRARY_PATH checked obviusly
